src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9972,100 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+/// Return the list of the commands in the specified buffer,
+/// if no buffer is specified, then return list of global(non-buffer) the
+/// commands
+///
+/// @param[in] buf the specified buffer, if NULL then global commands required
+///
+/// @return The array of the commands and the information of each command
+ArrayOf(Dictionary) commands_array(buf_T *buf)
+{
+  Array rv = ARRAY_DICT_INIT;
+  int i = 0, j = 0;
```

Comment:
please declare loop variables in the loop initializer when possible.

Reply:
ok that way is cool, but above some where they also did like it, so I thought this is the way we code like.

Reply:
BTW, done

Reply:
Yes, existing code from vim is different. But new code should follow this style (unless of course loop variable is used after loop)